### PR TITLE
Use revert endpoint for reverting service and add metadata only fetch

### PR DESCRIPTION
**Pull request by ryan-lane**
This change is aimed at returning less credential data, when that data is not necessary. Specifically, it's aiming to remove the return of full credential data when services are fetched through the interface, since those interfaces don't need the credential data.

Along with this change is a new method of doing reverts. I'm starting with service, because it's simpler, but future changes will also make this the method for credentials and blind_credentials, which will let us do reverts without needing to see values.

### CODE REVIEW

**Comment by ryan-lane** [comment]
I also found a bug in displaying conflicts in the history revert view, which I fixed by adding getCredentialByID and getBlindCredentialByID

File: confidant/public/modules/history/controllers/BlindCredentialHistoryCtrl.js [hunk 1]
```diff
@@ -69,6 +69,14 @@
                 }
             });
 
+            $scope.getCredentialByID = function(id) {
+                return $filter('filter')($scope.$parent.credentialList, {'id': id})[0];
```
Comment (skiptomyliu) [inline confidant/public/modules/history/controllers/BlindCredentialHistoryCtrl.js]: apologies, by JS is rusty.  We can always expect an entry to be returned by this?  Meaning, `[0]` is safe to return?

File: confidant/routes/v1.py [hunk 2]
```diff
@@ -324,18 +366,117 @@ def map_service_credentials(id):
...
+            revert_service.credentials,
+            revert_service.blind_credentials,
+        )
+        if conflicts:
+            ret = {
+                'error': 'Conflicting key pairs in mapped service.',
+                'conflicts': conflicts
+            }
+            return jsonify(ret), 400
+    # Try to save to the archive
```
Comment (skiptomyliu) [inline confidant/routes/v1.py]: Curious -- how come we save to archive here?   Is it a sort of backup?

File: confidant/public/modules/history/controllers/BlindCredentialHistoryCtrl.js (hunk 1 above)
Comment (ryan-lane) [reply confidant/public/modules/history/controllers/BlindCredentialHistoryCtrl.js]: `$scope.$parent.credentialList` is a pre-generated list. If there's no credentials in that list, then nothing would ever call `getCredentialByID`. If `getCredentialByID`, then something must be in the list.

File: confidant/routes/v1.py (hunk 2 above)
Comment (ryan-lane) [reply confidant/routes/v1.py]: Confidant's dynamo usage is a bit weird. We use a single table for all record types, and we store history and current records in the same table. For performance, we have two types of record per model. So here, we have `service` and `service-archive` type records. The `service` record is the one that's queried, when `get_service` is called. When history is looked up, it's the archive records.

If we were to split the records into separate tables, we would use hash+key, which would let us drop the double-save. It's something I've considered, if we were to get rid of blind credentials (which we can do with proper RBAC).

**Pull request merged**